public/app/features/explore/utils/query.ts

```diff
@@ -0,0 +1,81 @@
+import { DataQuery, AbstractQuery, AbstractLabelOperator, AbstractLabelMatcher } from '@grafana/data';
+import { Token } from 'prismjs';
+
+const FromPromLikeMap: Record<string, AbstractLabelOperator> = {
+  '=': AbstractLabelOperator.Equal,
+  '!=': AbstractLabelOperator.NotEqual,
+  '=~': AbstractLabelOperator.EqualRegEx,
+  '!~': AbstractLabelOperator.NotEqualRegEx,
+};
+const ToPromLikeMap: Record<AbstractLabelOperator, string> = {
+  [AbstractLabelOperator.Equal]: '=',
+  [AbstractLabelOperator.NotEqual]: '!=',
+  [AbstractLabelOperator.EqualRegEx]: '=~',
+  [AbstractLabelOperator.NotEqualRegEx]: '!~',
+};
+
+export function toPromLikeQuery(labelBasedQuery: AbstractQuery): PromLikeQuery {
```

Comment:
I would expect toPromLikeQuery to be in Prometheus code. If wee. want to also use it in Loki, we could import from Prometheus or duplicate. But I would not expect it here

Reply:
I've moved it to Prometheus code. I feel like it would be nice to have some shared code related to PromQL/LogQL language somewhere else in the future.